### ISSUE TITLE
Flaky tests: `assert False, (bad_thread, call_stacks)` - Worker executor thread still running after `Nanny.kill`

### ISSUE THREAD

**Issue by gjoseph92** [dask/distributed#6796]
- https://github.com/dask/distributed/issues/6794
- https://github.com/dask/distributed/issues/6795

In these tests, a Nanny is killed (either via `Nanny.kill`, or `Nanny.restart`, which calls `kill` internally), like:
https://github.com/dask/distributed/blob/a53858a5fc73b591da5a6ceff903a468990f8a21/distributed/deploy/tests/test_spec_cluster.py#L156-L158

It appears that in some cases the worker's ThreadPoolExecutor isn't being terminated? That's confusing though, since the worker should be in a subprocess, so its threads wouldn't even be visible to the parent process?

I have a feeling https://github.com/dask/distributed/pull/6427 would address this (or at least raise a meaningful error instead of this thread leakage?) because `Nanny.kill` doesn't actually wait for the subprocess to terminate.

**Comment by gjoseph92** [comment]
@hendrikmakait do you think this was fixed in https://github.com/dask/distributed/pull/6817? I'm still not understanding the mechanism for how (I know `test_localcluster_get_client` was leaking a thread, but why did that show up in the other tests instead?).

**Comment by hendrikmakait** [comment]
I looks like we stop seeing this flake, so from my perspective the symptom of flaking has been resolved. Let's add a follow-up issue to investigate the causal mechanism. It feels like there may be a more systemic issue in there, or at the very least, we might get a better understanding for unintended side effects of messy tests.

**Comment by hendrikmakait** [comment]
Closed as these tests stopped being flaky after #6817. Given known issues with the client (#5901, #6527), we have decided against further investigation at this time.
